src/modules/billing/services/batch-service.js

```diff
@@ -173,13 +172,17 @@ const setErrorStatus = async (batchId, errorCode) => {
 
 const approveBatch = async (batch, internalCallingUser) => {
   try {
-    await chargeModuleBillRunConnector.approve(batch.externalId);
-    await chargeModuleBillRunConnector.send(batch.externalId);
-
-    await saveEvent('billing-batch:approve', 'sent', internalCallingUser, batch);
-
-    await licencesService.updateIncludeInSupplementaryBillingStatusForSentBatch(batch.id);
-    await invoiceService.resetIsFlaggedForRebilling(batch.id);
+    await Promise.all([
```

Comment:
Just out of interest, why is this in a Promise.all()?

Might be worth leaving an inline code comment explaining what's happening here :)

Reply:
Note 1: You don't usually do `await`s inside the Promise.all() array.

Note 2: I would suggest constructing the array first and running a single Promise.all().

E.g. ```

const promises = [
chargeModuleBillRunConnector.approve(batch.externalId),
chargeModuleBillRunConnector.send(batch.externalId),
saveEvent('billing-batch:approve', 'sent', internalCallingUser, batch),
licencesService.updateIncludeInSupplementaryBillingStatusForSentBatch(batch.id),
invoiceService.resetIsFlaggedForRebilling(batch.id)
];

if (batch.type === BATCH_TYPE.supplementary) {
      promises.push(transactionsService.updateIsCredited(batch.region.id));
}

Promise.all(promises);

Reply:
Thanks yes an oversight on my behalf - half way to converting the code back after trying to debug some issues. I removed the promise all. Unless you think it is necessary to stay.